selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/ocp/OpenShiftLoginPage.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.selenium.pageobject.ocp;
+
+import static java.util.Arrays.asList;
+import static org.eclipse.che.selenium.pageobject.ocp.OpenShiftLoginPage.Locators.LOGIN_BUTTON_XPATH;
+import static org.eclipse.che.selenium.pageobject.ocp.OpenShiftLoginPage.Locators.PASSWORD_INPUT_NAME;
+import static org.eclipse.che.selenium.pageobject.ocp.OpenShiftLoginPage.Locators.USERNAME_INPUT_NAME;
+
+import com.google.inject.Inject;
+import org.eclipse.che.selenium.core.SeleniumWebDriver;
+import org.eclipse.che.selenium.core.webdriver.SeleniumWebDriverHelper;
+import org.eclipse.che.selenium.pageobject.site.LoginPage;
+import org.openqa.selenium.TimeoutException;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+
+public class OpenShiftLoginPage implements LoginPage {
+  private final SeleniumWebDriverHelper seleniumWebDriverHelper;
+
+  protected interface Locators {
+    String USERNAME_INPUT_NAME = "username";
+    String PASSWORD_INPUT_NAME = "password";
+    String LOGIN_BUTTON_XPATH = "//button[contains(text(),'Log In')]";
+  }
+
+  @FindBy(name = USERNAME_INPUT_NAME)
+  private WebElement usernameInput;
+
+  @FindBy(name = PASSWORD_INPUT_NAME)
+  private WebElement passwordInput;
+
+  @FindBy(xpath = LOGIN_BUTTON_XPATH)
+  private WebElement loginButton;
+
+  @Inject
+  public OpenShiftLoginPage(
+      SeleniumWebDriver seleniumWebDriver, SeleniumWebDriverHelper seleniumWebDriverHelper) {
+    this.seleniumWebDriverHelper = seleniumWebDriverHelper;
+
+    PageFactory.initElements(seleniumWebDriver, this);
+  }
+
+  @Override
+  public void login(String username, String password) {
+    waitOnOpen();
+
+    seleniumWebDriverHelper.setValue(usernameInput, username);
+    seleniumWebDriverHelper.setValue(passwordInput, password);
+    seleniumWebDriverHelper.waitAndClick(loginButton);
+
+    waitOnClose();
+  }
+
+  @Override
+  public boolean isOpened() {
```

Comment:
In my opinion, method "isOpened()" should check visibility in the current moment without any delays.
So maybe better to use something like:
```
  @Override
  public boolean isOpened() {
    return isVisible(usernameInput) && isVisible(passwordInput) && isVisible(loginButton);
  }

  private boolean isVisible(WebElement element) {
    return seleniumWebDriverHelper.isVisible(element);
  }
```
Or even more, add something like "SeleniumWebDriverHelper#isVisible(List<WebElement> elements)" method to the "SeleniumWebDriverHelper" class

Reply:
It's unexpected for me that you proposed to remove waiting on element from the page object method because earlier you are usually propagating an idea to make the tests more stable by enclosing check of readiness of web element to execute the action against it inside the page object methods.

Reply:
My point that "isOpened" method should check visibility without any delays
https://github.com/eclipse/che/blob/master/selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/SeleniumWebDriverHelper.java#L953-L965

Reply:
I understand your point. You are talking about instant verification.